src/neuron/container/soa_container.hpp

```diff
@@ -547,6 +571,37 @@ struct storage_info_impl: utils::storage_info {
     std::string m_container{}, m_field{};
     std::size_t m_size{};
 };
+
+class AccumulateMemoryUsage {
+  public:
+    void operator()(detail::index_column_tag const& indices,
+                    std::vector<detail::index_column_tag::type> const& vec,
+                    int field_index,
+                    int array_dim) const {
+        auto element_size = sizeof(detail::index_column_tag::type);
+
+        m_usage.stable_identifiers.size = vec.size() * (sizeof(vec[0]) + sizeof(size_t*));
+        m_usage.stable_identifiers.capacity = m_usage.stable_identifiers.size +
+                                              (vec.capacity() - vec.size()) * sizeof(vec[0]);
+    }
+
+    template <class Tag>
+    void operator()(Tag const& tag,
+                    std::vector<typename Tag::type> const& vec,
+                    int field_index,
+                    int array_dim) const {
+        m_usage.heavy_data += VectorMemoryUsage(vec);
+    }
+
+    StorageMemoryUsage usage() {
+        return m_usage;
+    }
+
+  private:
+    mutable StorageMemoryUsage m_usage;
```

Comment:
Can we remove `mutable` (🤢) by removing the `const`s on `operator()`?

Reply:
I may have missed something, but I though the callable is passed in as a const reference.

Reply:
Okay, so the solution is to rewrite those functions and use STL style accept by value and return the callable.

Reply:
I think the `mutable` and `const` (from `operator()`) can be removed now?